src/sentry/incidents/logic.py

```diff
@@ -591,6 +600,146 @@ class AlertRuleNameAlreadyUsedError(Exception):
 DEFAULT_ALERT_RULE_RESOLUTION = 1
 
 
+def validate_unified_alert_rule(data):
+    """Performs validation on an alert rule's POSTed data
+    This includes ensuring there is either 1 or 2 triggers, which each have actions, and have proper thresholds set.
+    The critical trigger should both alert and resolve 'after' the warning trigger. """
+    triggers = data.get("triggers", [])
+    if triggers:
+        if len(triggers) == 1:
+            if triggers[0]["label"] != CRITICAL_TRIGGER:
+                return False, "Trigger must be labeled 'CRITICAL'"
+        elif len(triggers) == 2:
+            if triggers[0]["label"] != CRITICAL_TRIGGER or triggers[1]["label"] != WARNING_TRIGGER:
+                return (
+                    False,
+                    "First trigger must be labeled 'CRITICAL', second trigger must be labeled 'WARNING'",
+                )
+            else:
+                # Verify thresholds. alertThreshold, resolveThreshold, threshholdType
+                if triggers[0]["thresholdType"] != triggers[1]["thresholdType"]:
+                    return False, "Must have matching threshold types (???)"
```

Comment:
Should `???` be replaced with something that will give the user more context on how to fix the validation error?

Reply:
Ah, I put it there because I wasn't sure if this was something to enforce. Now I am getting clarification on that.